### DISCUSSION TITLE
Issues with Deprecated Protocol Syntax and Version Mismatch in Nuclei

### DISCUSSION THREAD

**Discussion by aravindb26** [projectdiscovery/nuclei#3748] [Q&A]
Current Behavior:

I encountered the following issues while running Nuclei:

    Deprecated Protocol Syntax: The warning message mentions that one template (rails-debug-mode) is loaded with deprecated protocol syntax. It is advised to update the template to ensure continued support before version 2.9.5 of Nuclei.

    Templates and Nuclei-templates Version Mismatch: The output indicates that the current version of Nuclei is 2.9.4, while the current version of nuclei-templates is 9.5.0. It is recommended to use the latest versions of both Nuclei and nuclei-templates for compatibility and access to new features.

Expected Behavior:

I expected the following behavior:

    No warnings about deprecated protocol syntax in any templates.
    Consistency in the versions of Nuclei and nuclei-templates.

Steps To Reproduce:

To reproduce the issues, please follow these steps:

    Run the command nuclei -l list.txt -t ../nuclei-templates/exposures/logs/rails-debug-mode.yaml -v.
    Note the warning message about deprecated protocol syntax and the version mismatch between Nuclei and nuclei-templates.

Anything else:

I have added screenshots as proof of concept

**Comment by ehsandeep** [comment]
@aravindb26 see the pinned issue - https://github.com/projectdiscovery/nuclei/discussions/3648

**Comment by aravindb26** [comment]
@ehsandeep Yes i have updated version of nuclei already, the issue is with the nuclei-template

**Comment by ehsandeep** [comment]
@aravindb26 `nuclei -l list.txt -t http/exposures/logs/rails-debug-mode.yaml` is the correct way, also you can try using `nuclei -reset` option to avoid errors.

**Comment by aravindb26** [comment]
@ehsandeep I think u r wrong mate, -t flag indicates to indicate `nuclei-templates` which accepts to analyze the vulnerabilities.

I think http is wrong `typo`

**Comment by ehsandeep** [comment]
@aravindb26, please refer to link shared in the above comment, which includes information about the directory structure update in [nuclei-templates](https://github.com/projectdiscovery/nuclei-templates) project and all the information related to the error/warning you have been experiencing.

**Comment by aravindb26** [comment]
@ehsandeep It's an issue know? How it comes under discussion mate :)  ?

**Comment by ehsandeep** [comment]
it's a known issue with a documented solution - https://github.com/projectdiscovery/nuclei/discussions/3648

moved it to discuss use for others reference who might be facing similar issues, but this does not require any change in nuclei.

Let me know if you already used `nuclei -reset` option and still facing the issue, please paste a screenshot of the cmd you executed with error information.

**Comment by aravindb26** [comment]
@ehsandeep Yeah I have done that, it just simply removed my nuclei-templates.

I have reinstalled all templates and tried  but have the same issue again. So, i think it's an issue mate :)

**Reply by ehsandeep** [reply]
@aravindb26, please share the required information if you need any further assistance.

> please paste a screenshot of your executed cmd with the error information.

**Reply by aravindb26** [reply]
@ehsandeep Yeah I'm again providing here.
<img width="716" alt="nuc" src="https://github.com/projectdiscovery/nuclei/assets/89305482/8a7f1656-101d-4caa-b140-f0a26934def4">

**Reply by ehsandeep** [reply]
@aravindb26, thanks for confirming; sadly, you are still using the outdated template project and did not check the information I've shared multiple times.


Here is the template link that you are looking to use from the current / latest [template](https://github.com/projectdiscovery/nuclei-templates) project - 

```
https://github.com/projectdiscovery/nuclei-templates/blob/main/http/exposures/logs/rails-debug-mode.yaml
```
This is to showcase the latest changes includes `http` in the path and it's not a typo, since you are using the older path structure, you are getting the warning which is expected, this is not an issue.

**Reply by aravindb26** [reply]
@ehsandeep Yeah got it, but still we have access to old one too mate, better to remove it know 🙂https://github.com/projectdiscovery/nuclei-templates/tree/52f92b91a25a2672ff5bed2e9bba1d9761f31099

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
